### PR TITLE
chore: add organization search query to workspaces

### LINKED ISSUE #14444: Add `organizations` to workspace search query
The `/workspaces` page needs the ability to query by organization. This should copy the code from [audit logs](https://github.com/coder/coder/blob/a8c89a4a2f2cc196b05470bf4a8b2615efd40d3c/coderd/searchquery/search.go#L50-L69). The SQL query should only support `organization_id`, not by name. 

Ideally refactor this org code to be DRY for audit logs, templates, and workspacs.

### PR DESCRIPTION
**Pull request by Emyrk**
Closes https://github.com/coder/coder/issues/14444

### CODE REVIEW

File: coderd/searchquery/search.go [hunk 1]
```diff
@@ -95,7 +75,7 @@ func Users(query string) (database.GetUsersParams, []codersdk.ValidationError) {
 	return filter, parser.Errors
 }
 
-func Workspaces(query string, page codersdk.Pagination, agentInactiveDisconnectTimeout time.Duration) (database.GetWorkspacesParams, []codersdk.ValidationError) {
+func Workspaces(ctx context.Context, db database.Store, query string, page codersdk.Pagination, agentInactiveDisconnectTimeout time.Duration) (database.GetWorkspacesParams, []codersdk.ValidationError) {
```
Comment (f0ssel) [inline coderd/searchquery/search.go:78]: aside: I see why we do it in this package, but this is kind of a weird naming scheme for something that returns database query params.

File: coderd/searchquery/search_test.go [hunk 2]
```diff
@@ -243,7 +270,11 @@ func TestSearchWorkspace(t *testing.T) {
 		c := c
 		t.Run(c.Name, func(t *testing.T) {
 			t.Parallel()
-			values, errs := searchquery.Workspaces(c.Query, codersdk.Pagination{}, 0)
+			db := dbmem.New()
+			if c.Setup != nil {
+				c.Setup(t, db)
+			}
+			values, errs := searchquery.Workspaces(context.Background(), db, c.Query, codersdk.Pagination{}, 0)
```
Comment (f0ssel) [inline coderd/searchquery/search_test.go:277]: Can we not use the traditional way of getting a DB in tests so this can run against postgres too?

**Review by f0ssel — APPROVED** [review]

File: coderd/searchquery/search.go (hunk 1 above)
Comment (Emyrk) [reply coderd/searchquery/search.go:78]: Fair. Maybe it should be named `WorkspacesFilter` or something. I won't address here though

File: coderd/searchquery/search_test.go (hunk 2 above)
Comment (Emyrk) [reply coderd/searchquery/search_test.go:277]: I'd prefer not to for this. What would be even better is if I used the mock database, but I haven't messed with that at all yet.

**Pull request merged**